docs/serving/setting-up-custom-ingress-gateway.md

```diff
@@ -92,3 +79,63 @@ For the service above, it should be updated to:
 ```
 gateway.knative-serving.knative-ingress-gateway: custom-ingressgateway.custom-ns.svc.cluster.local
 ```
+
+## Replace the knative-ingress-gateway gateway
+
+We customized the gateway service so far, but we may also want to use our own gateway.
+We can replace the default gateway with our own gateway with following steps.
+
+#### Step 1: Create Gateway
+
+Let's say you replace the default `knative-ingress-gateway` gateway with
+`knative-custom-gateway` in `custom-ns`.
+First, we create the `knative-custom-gateway` gateway.
+
+```
+cat <<EOF | kubectl apply -f -
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: knative-custom-gateway
+  namespace: custom-ns
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+  - port:
+      number: 80
+      name: http
+      protocol: HTTP
+    hosts:
+    - "*"
+EOF
+```
+
+!!! note
+    Replace the label selector `istio: ingressgateway` with the label of your service.
+
+### Step 2: Update Gateway Configmap
+
+Update gateway configmap `config-istio` under `knative-serving`
+namespace:
+
+```bash
+kubectl edit configmap config-istio -n knative-serving
+```
+
+Replace the `gateway.knative-serving.knative-ingress-gateway` field with
+the customized gateway.
+
+```
+gateway.knative-serving.knative-ingress-gateway: "istio-ingressgateway.istio-system.svc.cluster.local"
+```
+
+For the gateway above, it should be updated to:
+
+```
+gateway.custom-ns.knative-custom-gateway: "istio-ingressgateway.istio-system.svc.cluster.local"
+```
+
+The configuration format should be `gateway.GATEWAY_NAMESPACE.GATEWAY_NAME`.
+The `GATEWAY_NAMESPACE` is optional. when it is omitted, the system will search for
```

Comment:
```suggestion
The configuration format should be `gateway.<gateway-namespace>.<gateway-name>`.
`<gateway-namespace>` is optional. When it is omitted, the system searches for
```

Reply:
Knative style rules say that placeholders should be in the format `<placeholder-name>` and not `PLACEHOLDER_NAME`.